client/core/trade.go

```diff
@@ -380,23 +393,12 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 		}
 	}
 
-	// Calculate the new filled value for the order.
-	isMarketBuy := t.Type() == order.MarketOrderType && !trade.Sell
-	var filled uint64
-	// If the order has been canceled, add that to the filled.
-	if t.cancel != nil && t.cancel.matches.maker != nil {
-		filled += t.cancel.matches.maker.Quantity
-	}
-	for _, n := range t.matches {
-		if isMarketBuy {
-			filled += calc.BaseToQuote(n.Match.Rate, n.Match.Quantity)
-		} else {
-			filled += n.Match.Quantity
-		}
-	}
-	// The filled amount includes all of the trackedTrade's matches, so the
-	// filled amount must be set, not just increased.
-	trade.SetFill(filled)
+	// NOTE: If Core was restarted, previously inactive matches are no longer in
+	// the t.matches map, preventing correct fill computation by summation of
+	// all match quantities for partially filled orders with inactive matches
+	// that had partially filled the order.
+	filled := trade.AddFill(newFill)
```

Comment:
More that will be obsolete with https://github.com/decred/dcrdex/pull/566 merged.